cmd/restic/cmd_version.go

```diff
@@ -21,8 +22,29 @@ Exit status is 0 if the command was successful, and non-zero if there was any er
 `,
 	DisableAutoGenTag: true,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Printf("restic %s compiled with %v on %v/%v\n",
-			version, runtime.Version(), runtime.GOOS, runtime.GOARCH)
+		if globalOptions.JSON {
+			type jsonVersion struct {
+				Version   string `json:"version"`
+				GoVersion string `json:"go_version"`
+				GoTarget  string `json:"go_target"`
+			}
+
+			jsonS := jsonVersion{
+				Version:   version,
+				GoVersion: runtime.Version(),
+				GoTarget:  runtime.GOOS + "/" + runtime.GOARCH,
+			}
+
+			err := json.NewEncoder(globalOptions.stdout).Encode(jsonS)
+			if err != nil {
+				Warnf("Encode failed: %v\n", err)
```

Comment:
The `diff` and `ls` commands output this error as `JSON encode failed: %v\n`. Can we please change it to that so it's more coherent (and also clear that it's about JSON)?